src/main/java/com/conveyal/gtfs/validator/PostLoadValidator.java

```diff
@@ -0,0 +1,135 @@
+            for (Rule rule : rules) {
+                if (rule.check(stop, feed)) break;
+            }
+        }
+    }
+
+    private static final String FILE = "stops.txt";
+    private static final String FIELD = "parent_station";
+
+    /** GTFS location_type codes. */
+    private enum LocType {
+        STOP(0),
+        STATION(1),
+        ENTRANCE(2),
+        GENERIC(3),
+        BOARDING(4);
+        public final int code;
+        LocType (int code) {
+            this.code = code;
+        }
+    }
+
+    private enum Requirement {
+        OPTIONAL, REQUIRED, FORBIDDEN
+    }
+
+    private static class Rule {
+        // The location_type to which the rule applies.
+        LocType fromLocType;
+        // Whether the parent_station is required, forbidden, or optional.
+        Requirement parentRequirement;
+        // If the parent_station is present and not forbidden, what location_type it must be.
+        LocType parentLocType;
+
+        public Rule (LocType fromLocType, Requirement parentRequirement, LocType parentLocType) {
+            this.fromLocType = fromLocType;
+            this.parentRequirement = parentRequirement;
+            this.parentLocType = parentLocType;
+        }
+
+        /**
+         * Call this method once for each location_type, defining a rules about that location_type's parent_station.
+         * @return true if this rule validated the stop, or false if more rules must be checked
+         */
+        public boolean check (Stop stop, GTFSFeed feed) {
+            // If this rule does not apply to this stop, do not perform checks.
+            if (stop.location_type != fromLocType.code) {
+                return false;
+            }
+            if (stop.parent_station == null) {
+                if (parentRequirement == Requirement.REQUIRED) {
+                    feed.errors.add(new EmptyFieldError(FILE, stop.sourceFileLine, FIELD));
+                }
+            } else {
+                // Parent station reference was supplied.
+                if (parentRequirement == Requirement.FORBIDDEN) {
+                    feed.errors.add(new ForbiddenFieldError(FILE, stop.sourceFileLine, FIELD));
+                }
+                Stop parentStation = feed.stops.get(stop.parent_station);
+                if (parentStation == null) {
+                    feed.errors.add(new ReferentialIntegrityError(FILE, stop.sourceFileLine, FIELD, stop.parent_station));
+                } else {
+                    if (parentStation.location_type != parentLocType.code) {
+                        feed.errors.add(new RangeError(
+                                FILE, stop.sourceFileLine, FIELD,
+                                parentLocType.code, parentLocType.code,
+                                parentStation.location_type)
+                        );
+                    }
+                }
+            }
+            return true;
+        }
+    }
+
+    private static final List<Rule> rules = List.of(
```

Comment:
Similar to comment above -- I initially thought this specified rules that applied to validation methods beyond `validateStopStationConstraints`